### PR TITLE
feat: Adding nodes

### PR DESCRIPTION
**Pull request by joshuef**
- Add --add cli flag for adding nodes to an existing network easily
- add env fallback for node count if available

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -305,19 +313,46 @@ pub fn run_with(cmd_args: Option<&[&str]>) -> Result<(), String> {
...
+
+    let existing_nodes_count = paths
+        .collect::<Result<Vec<_>, io::Error>>()
+        .map_err(|_| "Error collecting testnet log dir".to_string())?
+        .len();
+
+    println!("existing nodes: {:?}", existing_nodes_count);
+
+    // either we have genesis only, or all existing ndoes
+    let start = if existing_nodes_count == 0 {
```
Comment (Yoga07) [inline src/lib.rs:327]: Should this be `== 1`? Because the genesis node would be having its directory.

Comment (joshuef) [reply src/lib.rs:327]: hmm, yeh this should not be hittable... 🤔  let me try removing it

File: src/lib.rs [hunk 2]
```diff
@@ -305,19 +313,43 @@ pub fn run_with(cmd_args: Option<&[&str]>) -> Result<(), String> {
...
+
+    let existing_nodes_count = paths
+        .collect::<Result<Vec<_>, io::Error>>()
+        .map_err(|_| "Error collecting testnet log dir".to_string())?
+        .len();
+
+    println!("{:?} existing nodes found", existing_nodes_count);
+
+    // either we have genesis only, or all existing ndoes
+    let start = existing_nodes_count;
```
Comment (Yoga07) [inline src/lib.rs:327]: Would it be better to bail `if existing_nodes_count == 0 && adding_nodes` ? Meaning there's no genesis to add nodes to.

Comment (joshuef) [reply src/lib.rs]: bail sure. i wouldn't try again, if the first failed, there's something bigger afoot

**Review by Yoga07 — APPROVED** [review]

**Pull request merged**